wasm/func_validation_test.go

```diff
@@ -0,0 +1,38 @@
+package wasm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateFunction_valueStackLimit(t *testing.T) {
+	const max = 100
+	const valuesNum = max + 1
+
+	// Build a function which has max+1 const instructions.
+	f := &FunctionInstance{Body: []byte{}, FunctionType: &TypeInstance{Type: &FunctionType{}}}
+	for i := 0; i < valuesNum; i++ {
+		f.Body = append(f.Body, OpcodeI32Const, 1)
+	}
+
+	// Drop all the consts so that if the max is higher, this function body would be sound.
+	for i := 0; i < valuesNum; i++ {
+		f.Body = append(f.Body, OpcodeDrop)
+	}
+
+	// Plus all functions must end with End opcode.
+	f.Body = append(f.Body, OpcodeEnd)
+
+	t.Run("not exceed", func(t *testing.T) {
+		err := validateFunctionInstance(f, nil, nil, nil, nil, nil, max+1)
+		require.NoError(t, err)
+	})
+	t.Run("exceed", func(t *testing.T) {
+		err := validateFunctionInstance(f, nil, nil, nil, nil, nil, max)
+		require.Error(t, err)
+		expMsg := fmt.Sprintf("function too large: potentially could have %d values on the stack with the limit %d", valuesNum, max)
```

Comment:
is there a similar message somewhere from another tool that crashes when there are too many stack values?

I would suggest adding the function name or index to the message.

Also, "potentially" and "could have" are both passive. maybe "function[%d] may have %d stack values, which  exceeds limit %d"